### PR TITLE
Upgrade pandas and update dependency version range

### LINKED ISSUE #80: Upgrade Pandas version to latest version >=0.23.1
I love marbles! I also enjoyed Jane's talk at PyData Amsterdam.

My issue is that installing marbles automatically downgrades your pandas version.

If you do: `pip install pandas` it installs version 0.23.1 at the time of writing.
Then running `pip install marbles` downgrades pandas to 0.21.

This causes the following code to break:

```
from datetime import datetime
import pandas as pd

data = {
            datetime(2018,4,1): [2500, 1500],
            datetime(2018,5,1): [2000, 2500],
            datetime(2018,6,1): [2000, 1500],
    }
df = pd.DataFrame.from_dict(data, orient='index', columns = ['Income', 'Expense'])
print(df)
```

I believe the line is on marbles/marbles/mixins/setup.py line 57. 

I could try and do a pull request if you wanted, but I'm not sure what all tests I would need to ensure are passing?

### PR DESCRIPTION
**Pull request by thejunglejane**
Closes #80

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -6,6 +6,7 @@ name = "pypi"
 [packages]
 "2faa170" = {path = "./marbles/core", extras = ["dev"], editable = true}
 "55bbcad" = {path = "./marbles/mixins", extras = ["dev"], editable = true}
+pandas = "==0.23.1"
```
Comment (thejunglejane) [inline Pipfile]: I feel like we don't want this?

Comment (leifwalsh) [reply Pipfile]: How did this get in? Did you run something like `pipenv install pandas==0.23.1`?

Comment (leifwalsh) [reply Pipfile]: The `Pipfile` shouldn't reference packages that `marbles.mixins` depends on, generally. We should probably just delete this line, but I want to understand how it got in.

Comment (thejunglejane) [reply Pipfile]: I did indeed do that

Comment (leifwalsh) [reply Pipfile]: I'm not sure. I think the right thing to do is to update our `setup.py` files and then run `pipenv lock` to update the lockfile.

**Pull request merged**